File: VICSApp/CoreSystem/DHCD/FormUI/Report/ReportViewer.vb
Imports CrystalDecisions.CrystalReports.Engine

Partial Public Class ReportViewer
    Public Sub New()
        InitializeComponent()
    End Sub

    Public Sub New(reportDocument As ReportClass)
        Me.New()
        CrystalReportViewer1.ReportSource = reportDocument
    End Sub


    Private Sub ReportViewer_Load(sender As Object, e As EventArgs)
        CrystalReportViewer1.Show()
    End Sub

    Friend Shared Function LoadReport(reportDocument As ReportClass, owner As Form) As ReportViewer
        Dim v As New ReportViewer(reportDocument)
        v.Show()
        v.WindowState = FormWindowState.Maximized
        Return v
    End Function
End Class

File: VICSApp/CoreSystem/DHCD/FormUI/Report/ReportViewer.cs
using System;
using System.Windows.Forms;
using CrystalDecisions.CrystalReports.Engine;

namespace pmDHCD
{
    public partial class ReportViewer
    {
        public ReportViewer()
        {
            InitializeComponent();
        }

        public ReportViewer(ReportClass reportDocument) : this()
        {
            CrystalReportViewer1.ReportSource = reportDocument;
        }

        private void ReportViewer_Load(object sender, EventArgs e)
        {
            CrystalReportViewer1.Show();
        }

        internal static ReportViewer LoadReport(ReportClass reportDocument, Form owner)
        {
            var v = new ReportViewer(reportDocument);
            v.Show();
            v.WindowState = FormWindowState.Maximized;
            return v;
        }
    }
}